[heos] `NullPointerException` during startup

I'm not quite sure if this bug is in the binding or in core, but I've recently seen this callstack a few times in my logs. This is when starting openHAB:

```text
2026-01-11 21:16:21.754 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.NullPointerException: null
	at java.util.Objects.requireNonNull(Unknown Source) ~[?:?]
	at java.util.ImmutableCollections$MapN.<init>(Unknown Source) ~[?:?]
	at java.util.Map.ofEntries(Unknown Source) ~[?:?]
	at java.util.Map.copyOf(Unknown Source) ~[?:?]
	at org.openhab.core.thing.Channel.getProperties(Channel.java:166) ~[?:?]
	at org.openhab.core.thing.dto.ChannelDTOMapper.map(ChannelDTOMapper.java:43) ~[?:?]
	at org.openhab.core.thing.dto.ThingDTOMapper.map(ThingDTOMapper.java:49) ~[?:?]
	at org.openhab.core.thing.events.ThingEventFactory.map(ThingEventFactory.java:491) ~[?:?]
	at org.openhab.core.thing.events.ThingEventFactory.createUpdateEvent(ThingEventFactory.java:468) ~[?:?]
	at org.openhab.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:146) ~[?:?]
	at org.openhab.core.thing.internal.ThingRegistryImpl.notifyListenersAboutUpdatedElement(ThingRegistryImpl.java:1) ~[?:?]
	at org.openhab.core.common.registry.AbstractRegistry.updated(AbstractRegistry.java:334) ~[?:?]
	at org.openhab.core.thing.internal.ThingManagerImpl.thingUpdated(ThingManagerImpl.java:254) ~[?:?]
	at org.openhab.core.thing.internal.ThingHandlerCallbackImpl.thingUpdated(ThingHandlerCallbackImpl.java:156) ~[?:?]
	at org.openhab.core.thing.binding.BaseThingHandler.updateThing(BaseThingHandler.java:490) ~[?:?]
	at org.openhab.binding.heos.internal.handler.HeosBridgeHandler.updateThingChannels(HeosBridgeHandler.java:427) ~[?:?]
	at org.openhab.binding.heos.internal.handler.HeosBridgeHandler.addPlayerChannel(HeosBridgeHandler.java:369) ~[?:?]
	at org.openhab.binding.heos.internal.handler.HeosBridgeHandler.lambda$4(HeosBridgeHandler.java:305) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
```

## Expected Behavior

No `NullPointerException` thrown.

## Current Behavior

The opposite of expected: `NullPointerException` thrown.

## Possible Solution
<!-- Not obligatory, but suggest a fix/reason for the bug, -->
<!-- or ideas how to implement the addition or change -->

## Steps to Reproduce (for Bugs)

Not sure yet, will keep an eye on it and try to spot a pattern.

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version used: 5.1.1
* Environment name and version (e.g. Chrome 76, Java 8, Node.js 12.9, ...):
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...):

I finally got a proper stack trace thanks to openhab/openhab-core#5416:

```text
2026-04-03 16:38:35.231 [ERROR] [thing.binding.builder.ChannelBuilder] - Unexpected properties (pid) with null key or value for channel heos:bridge:f83fc6bf-4671-1aa8-0080-000678a4d33f:Pdenon; probably a bug in the related binding!
2026-04-03 16:38:35.231 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.IllegalArgumentException: Unexpected properties (pid) with null key or value for channel heos:bridge:f83fc6bf-4671-1aa8-0080-000678a4d33f:Pdenon
	at org.openhab.core.thing.binding.builder.ChannelBuilder.withProperties(ChannelBuilder.java:152)
	at org.openhab.binding.heos.internal.handler.HeosBridgeHandler.addPlayerChannel(HeosBridgeHandler.java:368)
	at org.openhab.binding.heos.internal.handler.HeosBridgeHandler.lambda$4(HeosBridgeHandler.java:305)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

--- a/bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosBridgeHandler.java
+++ b/bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosBridgeHandler.java
@@ -219,7 +219,7 @@ public class HeosBridgeHandler extends BaseBridgeHandler implements HeosEventLis
                     }
                 }
             } catch (HeosNotFoundException e) {
-                logger.debug("SKipping handler which reported not found", e);
+                logger.debug("Skipping handler which reported not found", e);
             }
         }
     }

--- a/bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosGroupHandler.java
+++ b/bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosGroupHandler.java
@@ -122,11 +122,11 @@ public class HeosGroupHandler extends HeosThingBaseHandler {
 
     @Override
     public String getId() throws HeosNotFoundException {
-        String localGroupId = this.gid;
-        if (localGroupId == null) {
+        String gid = this.gid;
+        if (gid == null) {
             throw new HeosNotFoundException();
         }
-        return localGroupId;
+        return gid;
     }
 
     public String getGroupMemberHash() {
@@ -155,10 +155,10 @@ public class HeosGroupHandler extends HeosThingBaseHandler {
             return;
         }
 
-        String localGid = this.gid;
+        String gid = this.gid;
         String eventGroupId = eventObject.getAttribute(HeosCommunicationAttribute.GROUP_ID);
         String eventPlayerId = eventObject.getAttribute(HeosCommunicationAttribute.PLAYER_ID);
-        if (localGid == null || !(localGid.equals(eventGroupId) || localGid.equals(eventPlayerId))) {
+        if (gid == null || !(gid.equals(eventGroupId) || gid.equals(eventPlayerId))) {
             return;
         }
 

--- a/bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosPlayerHandler.java
+++ b/bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosPlayerHandler.java
@@ -26,6 +26,7 @@ import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.heos.internal.configuration.PlayerConfiguration;
 import org.openhab.binding.heos.internal.exception.HeosFunctionalException;
+import org.openhab.binding.heos.internal.exception.HeosNotFoundException;
 import org.openhab.binding.heos.internal.json.dto.HeosErrorCode;
 import org.openhab.binding.heos.internal.json.dto.HeosEventObject;
 import org.openhab.binding.heos.internal.json.dto.HeosResponseObject;
@@ -51,7 +52,7 @@ import org.slf4j.LoggerFactory;
 public class HeosPlayerHandler extends HeosThingBaseHandler {
     private final Logger logger = LoggerFactory.getLogger(HeosPlayerHandler.class);
 
-    private @NonNullByDefault({}) String pid;
+    private @Nullable String pid;
     private @Nullable Future<?> scheduledFuture;
 
     public HeosPlayerHandler(Thing thing, HeosDynamicStateDescriptionProvider heosDynamicStateDescriptionProvider) {
@@ -83,6 +84,11 @@ public class HeosPlayerHandler extends HeosThingBaseHandler {
     }
 
     private synchronized void delayedInitialize() {
+        String pid = this.pid;
+        if (pid == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Player ID is missing");
+            return;
+        }
         try {
             refreshPlayState(pid);
 
@@ -117,7 +123,11 @@ public class HeosPlayerHandler extends HeosThingBaseHandler {
     }
 
     @Override
-    public String getId() {
+    public String getId() throws HeosNotFoundException {
+        String pid = this.pid;
+        if (pid == null) {
+            throw new HeosNotFoundException();
+        }
         return pid;
     }
 
@@ -127,7 +137,8 @@ public class HeosPlayerHandler extends HeosThingBaseHandler {
 
     @Override
     public void playerStateChangeEvent(HeosEventObject eventObject) {
-        if (!pid.equals(eventObject.getAttribute(PLAYER_ID))) {
+        String pid = this.pid;
+        if (pid == null || !pid.equals(eventObject.getAttribute(PLAYER_ID))) {
             return;
         }
 
@@ -141,7 +152,8 @@ public class HeosPlayerHandler extends HeosThingBaseHandler {
 
     @Override
     public void playerStateChangeEvent(HeosResponseObject<?> responseObject) throws HeosFunctionalException {
-        if (!pid.equals(responseObject.getAttribute(PLAYER_ID))) {
+        String pid = this.pid;
+        if (pid == null || !pid.equals(responseObject.getAttribute(PLAYER_ID))) {
             return;
         }
 
@@ -149,8 +161,8 @@ public class HeosPlayerHandler extends HeosThingBaseHandler {
     }
 
     @Override
-    public void playerMediaChangeEvent(String eventPid, Media media) {
-        if (!pid.equals(eventPid)) {
+    public void playerMediaChangeEvent(String pid, Media media) {
+        if (!pid.equals(this.pid)) {
             return;
         }
 

